shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+                cookieCfg.ignoreCookies = ["ignoredCookie"];
+                
+                let manager = createCookieMgr({
+                    cookieCfg: cookieCfg
+                });
+
+                // Start with cookies disabled
+                manager.setEnabled(false);
+
+                let ignoredKey = "ignoredCookie";
+                let allowedKey = "allowedCookie";
+                let newValue = newId();
+                
+                // Try to set ignored cookie while disabled - should not be cached
+                let ignoredResult = manager.set(ignoredKey, newValue);
+                Assert.equal(false, ignoredResult, "Setting ignored cookie should return false");
+                Assert.equal("", manager.get(ignoredKey), "Should not return cached value for ignored cookie");
+                
+                // Set allowed cookie while disabled - should be cached
+                let allowedResult = manager.set(allowedKey, newValue);
+                Assert.equal(true, allowedResult, "Setting allowed cookie should return true");
+                Assert.equal(newValue, manager.get(allowedKey), "Should return cached value for allowed cookie");
+
+                // Enable cookies - only allowed cookie should be flushed
+                manager.setEnabled(true);
+                
+                Assert.equal(undefined, this._testCookies[ignoredKey], "Ignored cookie should not be in storage");
+                Assert.equal(newValue + "; path=/", this._testCookies[allowedKey], "Allowed cookie should be in storage");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - multiple enable/disable cycles",
+            test: () => {
+                let manager = createCookieMgr(this._config);
+
+                let newKey1 = "test1." + newId();
+                let newKey2 = "test2." + newId();
+                let newValue1 = newId();
+                let newValue2 = newId();
+
+                // First cycle: disable, set, enable
+                manager.setEnabled(false);
+                manager.set(newKey1, newValue1);
+                Assert.equal(newValue1, manager.get(newKey1), "Should return cached value");
+                
+                manager.setEnabled(true);
+                Assert.equal(newValue1 + "; path=/", this._testCookies[newKey1], "First cookie should be flushed");
+
+                // Second cycle: disable again, set different cookie, enable
+                manager.setEnabled(false);
+                manager.set(newKey2, newValue2);
+                Assert.equal(newValue2, manager.get(newKey2), "Should return second cached value");
+                Assert.equal(newValue1, manager.get(newKey1), "Should still return first value from storage");
+                
+                manager.setEnabled(true);
+                Assert.equal(newValue2 + "; path=/", this._testCookies[newKey2], "Second cookie should be flushed");
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect